apps/discovery-platform/src/entry-server/routers/v1/index.ts

```diff
@@ -495,5 +510,161 @@ export const v1Router = (ops: {
+        }
+
+        if (!client) {
+          return res
+            .status(400)
+            .json({ errors: "client was not sent in request" });
+        }
+
+        let dbClient = await getClient(ops.db, client);
+
+        const clientIsTrialMode =
+          dbClient?.payment === constants.TRIAL_PAYMENT_MODE;
+        // set who is going to pay for quota
+        const paidById = clientIsTrialMode
+          ? constants.TRIAL_CLIENT_ID
+          : dbClient?.id;
+
+        // TODO: check if client has enough quota
+
+        // using the availability monitor results
+        // we create a object keyed by WORKING entry nodes
+        // linked with WORKING exit nodes which availability-monitor
+        // has proven connectivity between them
+        const routes = Array.from(
+          ops.getAvailabilityMonitorResults().entries()
+        ).reduce<{ entryNodePeerId: string; exitNodePeerIds: string[] }[]>(
+          (result, [entryNodePeerId, info]) => {
+            const { outgoingChannels, exitNodesToOutgoingChannels } =
+              info.connectivityReview;
+            const intermediatePeerIds: string[] = [];
+            const exitNodePeerIds: string[] = [];
+
+            // entry nodes must be stable
+            if (info.isStable && !excludeList.includes(entryNodePeerId)) {
+              // find intermediate nodes with working PING
+              for (const [intermediatePeerId, ping] of Object.entries(
+                outgoingChannels
+              )) {
+                if (ping > 0) intermediatePeerIds.push(intermediatePeerId);
+              }
+
+              // find exit nodes which have working PING with one of the intermediate nodes
+              for (const [exitNodePeerId, outgoingChannels] of Object.entries(
+                exitNodesToOutgoingChannels
+              )) {
+                for (const [intermediatePeerId, ping] of Object.entries(
+                  outgoingChannels
+                )) {
+                  if (
+                    intermediatePeerIds.includes(intermediatePeerId) &&
+                    ping > 0 &&
+                    !exitNodePeerIds.includes(exitNodePeerId)
+                  ) {
+                    exitNodePeerIds.push(exitNodePeerId);
+                  }
+                }
+              }
+
+              // update result if we have found working routes
+              if (exitNodePeerIds.length > 0) {
+                result.push({ entryNodePeerId, exitNodePeerIds });
+              }
+            }
+
+            return result;
+          },
+          []
+        );
+
+        // return 404 if we didn't find enough working nodes
+        if (routes.length < amount) {
+          return res.status(404).json({
+            errors: `client requested ${amount} routes but only found ${routes.length}`,
+          });
+        }
+
+        // get a random selection of routes
+        const selectedRoutes = routes
+          .sort(() => 0.5 - Math.random())
```

Comment:
Why `0.5 - Math.random()` instead of just `Math.random()`?

Reply:
because we want it to distribute negative numbers as to shuffle the array - certainly not the most entropic way of doing it!

Reply:
sort will actually first convert that number into a string and then compare. which means the result would not matter if it is negative or positive, just that it is a different number/string.

Reply:
isnt that only happening when no comparator is supplied? AKA the default comparator - what would you do alternatively here?

Reply:
If you only want to shuffle the array, just go with `Math.random()` instead of `0.5 - Math.random()`

Reply:
You're right. Just go with `0.5 - Math.random()` then. 
Since `Math.random` will always be positive, the default comparison does not randomize at all.